graph/pull_v2.go

```diff
@@ -223,6 +223,9 @@ func (p *v2Puller) pullV2Tag(tag, taggedName string) (verified bool, err error)
 	go func() {
 		if _, err := io.Copy(out, pipeReader); err != nil {
 			logrus.Errorf("error copying from layer download progress reader: %s", err)
+			if _, err := io.Copy(ioutil.Discard, pipeReader); err != nil {
```

Comment:
This pipe cancellation technique was supposed to be done on the download stream, not the progress channel.

Either way, `pipeReader` isn't connected to anything.